### ISSUE TITLE
[DFHack 0.42.04-alpha1] exportlegends.lua generates invalid XML structure in rare cases

### ISSUE THREAD

**Issue by Kromtec** [DFHack/dfhack#782]
I created some worlds where everything went fine :+1: , but in some rare cases the exportlegends script writes invalid xml child nodes into legends_plus.xml.

Like this one in `<group></group>`:
```XML
	<historical_event>
		<id>175425</id>
		<type>hist_figure_reach_summit</type>
		<group><vector<int32_t>: 0x2e2aced0></group>
	</historical_event>
```

**Comment by lethosor** [comment]
Can you upload the save?

**Comment by Kromtec** [comment]
Sure, here is a [Save](http://dffd.bay12games.com/file.php?id=11595) with this bug.

**Comment by lethosor** [comment]
I don't recall if I tested that save, but does this occur in alpha2? I changed some stuff with I/O in exportlegends recently, which could have caused it to write things to the xml file that should have gone to the console.

**Comment by Kromtec** [comment]
I get the same error in alpha2.

**Comment by lethosor** [comment]
Can you confirm whether this was fixed in #793?

**Comment by Kromtec** [comment]
Yes, #793 fixed it! :+1:
